src/sympc/applications/sort.py

```diff
@@ -0,0 +1,74 @@
+"""Applications that make use of primitive MPC operations."""
+
+# stdlib
+from typing import List
+
+# third party
+# third-party
+import torch
+
+from sympc.tensor import MPCTensor
+
+
+def sort(input_list: List[MPCTensor], ascending: bool = True) -> List[MPCTensor]:
+    """Takes a list of MPCTensors and sorts them in ascending/desending order.
+
+    Args:
+        input_list (List[MPCTensor]): Takes a list of MPCTensor
+        ascending (bool): If list has to sorted in ascending/descending order
+
+    Returns:
+        List[MPC_tensor]: Sorted list of MPCTensors
+    """
+
+    # Checks if the list of MPCTensors are of length 1
+    if not all((item.shape == torch.Size([1])) for item in input_list):
+
+        raise ValueError(
+            "Invalid dimension. All MPCTensors should have an 1-dimensional secret."
+        )
+
+    if len(input_list) > 1:
+        mid = len(input_list) // 2
+        left = input_list[:mid]
+        right = input_list[mid:]
+
+        # Recursive call on each half
+        sort(left)
+        sort(right)
+
+        # Two iterators for traversing the two halves
+        i = 0
+        j = 0
+
+        # Iterator for the main list
+        k = 0
+
+        while i < len(left) and j < len(right):
+            if (left[i] < right[j]).reconstruct():
```

Comment:
Q: Reconstructing here doesn't reveal information about the values? (to the orchestrater at least?)
Shouldn't we use ```*``` as a "selection"/masking mechanism?
If you had a tensor with only 2 elements to find the first one (ascending order).
```
bigger_right = left < right # Still hidden
first_element = (bigger_right - 1) * right + bigger_right * left
```
^ This is not the most efficient, but I think it is sufficient to showcase how we can interchange two elements?
@LaRiffle

Reply:
Yes I agree we should not reconstruct any thing, @gmuraru is a good solution to avoid this!

Reply:
I thought the same about it leaking too many values. But, I didn't know any other approach. I will look into the solution you proposed. Thank you.

Reply:
I have incorporated what @gmuraru had suggested. But, for now, it involves a Bubblesort implementation.